### PR TITLE
Remove _arrAlias field

### PR DESCRIPTION
**Pull request by benharsh**
The _arrAlias field was created when we still implemented slices in
closed-form. With the advent of ArrayViews, this is no longer necessary
and _arrAlias was basically only used to determine if we needed to
decrement the reference counts for array elements that are arrays.

This commit replaces the _arrAlias field with a bool field named
'_decEltRefCounts'. This field is set when the utility function
chpl_incRefCountsForDomainsInArrayEltTypes is called. The _delete_arr
function in ChapelDistribution will now check this field to determine if
we need to call chpl_decRefCountsForDomainsInArrayEltTypes.

Consequently, the 'isslice' bool formal of the 'dsiDestroyArr' method is
no longer necessary.

Testing:
- [x] full local
- [x] full gasnet

### CODE REVIEW

**Comment by benharsh** [comment]
@bradcray : how does this look?

File: modules/dists/BlockDist.chpl [hunk 1]
```diff
@@ -1288,10 +1288,6 @@ proc BlockArr.doiCanBulkTransfer(viewDom) {
     for param i in 1..rank do
       if viewDom.dim(i).stride != 1 then return false;
 
-  // See above note regarding aliased arrays
```
Comment (mppf) [inline modules/dists/BlockDist.chpl]: Is there a note above to remove as well?

Comment (benharsh) [reply modules/dists/BlockDist.chpl]: There is a config const that could be removed called `disableAliasedBulkTransfer`. It's only used in a couple of compopts and in miniMD's release Makefile. Removing it with this PR would not change the behavior of those programs.

Comment (mppf) [reply modules/dists/BlockDist.chpl]: Seems like we should remove it soon though, right?

Comment (benharsh) [reply modules/dists/BlockDist.chpl]: I believe we should, yes. Any objections?

Comment (mppf) [reply modules/dists/BlockDist.chpl]: I'm happy with removing it, if it is no longer used after this PR. But you could leave that to another PR if you like.

Comment (benharsh) [reply modules/dists/BlockDist.chpl]: I'll leave it for another PR for now, and give @bradcray a chance to chime in.

**Pull request merged**